src/platform/Zephyr/BLEAdvertisingArbiter.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "BLEAdvertisingArbiter.h"
+
+#include <lib/support/CodeUtils.h>
+#include <system/SystemError.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace {
+
+const BLEAdvertisingArbiter::Request & ToRequest(const sys_snode_t * node)
+{
+    return *static_cast<const BLEAdvertisingArbiter::Request *>(node);
+}
+
+} // namespace
+
+CHIP_ERROR BLEAdvertisingArbiter::RequestAdvertising(Request & request)
+{
+    sys_snode_t * prev = nullptr;
+    sys_snode_t * node;
```

Comment:
Why not initialize this with `nullptr` as well?

Reply:
It's only used in the loop and the loop-macro takes care of assigning a value to it before each iteration, so initialization is technically unnecessary, but I can add it to clear doubts.